elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -263,12 +306,319 @@ private Object fetchRelationship(Environment context) {
+        Map<String, Object> input = stripRelationships(dataField, entityClass, requestScope.getDictionary());
+
+        Optional<PersistentResource> fetchEntry = fetchObject(requestScope, entityClass,
+                        Optional.of(Arrays.asList(id)), Optional.empty(),
+                        Optional.empty(), Optional.empty(), Optional.empty()).stream().findFirst();
+        return fetchEntry
+                .map(persistentResource -> updateAttributes(persistentResource, entityClass, input, requestScope))
+                .orElseGet(() -> createObject(input, fieldName, requestScope, Optional.of(id), parent));
+    }
+
+    /**
+     * Strips out the attributes from the {@code element} singleton list and returns just the relationships, if present.
+     * @param element data input
+     * @param entityClass entity class
+     * @param dictionary entity dictionary
+     * @return relationship map
+     */
+    private static Map<String, Object> stripAttributes(Map<String, Object> element, Class<?> entityClass,
+                                                EntityDictionary dictionary) {
+        Map<String, Object> relationshipsOnly = new HashMap<>();
+
+        for(String key : element.keySet()) {
+            if(dictionary.isRelation(entityClass, key)) {
+                relationshipsOnly.put(key, element.get(key));
+            }
+        }
+        return relationshipsOnly;
+    }
 
-    private Object upsertObjects(Environment context) { return null; }
+    /**
+     * Strips out the relationships from the {@code element} singleton list and returns just the attributes, if present.
+     * @param element data input
+     * @param entityClass entity class
+     * @param dictionary entity dictionary
+     * @return attributes map
+     */
+    private static Map<String, Object> stripRelationships(Map<String, Object> element, Class<?> entityClass,
+                                                   EntityDictionary dictionary) {
+        Map<String, Object> attributesOnly = new HashMap<>();
+        String idFieldName = getIdFieldName(dictionary, entityClass);
+
+        for(String key : element.keySet()) {
+            if(dictionary.isAttribute(entityClass, key)) {
+                attributesOnly.put(key, element.get(key));
+            }
+            if(Objects.equals(key, idFieldName)) {
+                attributesOnly.put(key, element.get(key));
+            }
+        }
+        return attributesOnly;
+    }
 
-    private Object removeObjects(Environment context) { return null; }
+    /**
+     * Creates a new "root-level" resource in the database
+     * @param input individual input data objects
+     * @param fieldName graphql-java field type
+     * @param requestScope request scope
+     * @param uuid uuid
+     * @return {@link PersistentResource} object
+     */
+    private PersistentResource createObject(Map<String, Object> input, String fieldName,
```

Comment:
This function should be called createAndUpdate since it does both.  Either that or don't do the update here (instead call the update elsewhere in the calling code).